File: react-native/client/ios/Records/RequestRecord.swift
import CoinbaseWalletSDK
import ExpoModulesCore
import Foundation

struct RequestRecord : Record {
    @Field
    var actions: [ActionRecord]

    @Field
    var account: AccountRecord?
}

extension RequestRecord {
    var asRequest: Request {
        let requestActions: [Action] = self.actions.map { $0.asAction }
        let requestAccount: Account? = self.account?.asAccount
        return Request(actions: requestActions, account: requestAccount)
    }
}

File: react-native/client/android/src/main/java/expo/modules/coinbasewalletsdkexpo/records/RequestRecord.kt
package expo.modules.coinbasewalletsdkexpo.records

import com.coinbase.android.nativesdk.message.request.RequestContent
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class RequestRecord : Record {
    @Field
    var actions: List<ActionRecord> = listOf()

    @Field
    var account: AccountRecord? = null
}

val RequestRecord.asRequest: RequestContent.Request
    get() {
        return RequestContent.Request(
            actions = this.actions.map { it.asAction },
            account = this.account?.asAccount
        )
    }